examples/basic/evm-example/src/components/SwapContent/SwapContent.tsx

```diff
@@ -0,0 +1,599 @@
+        referrerAddress: null,
+        referrerFee: null,
+        slippage: '1.0',
+        swapperGroups: disabledLiquiditySources,
+        swappersGroupsExclude: true,
+      }
+      if (testMessagePassing) {
+        swapRequest = {
+          ...swapRequest,
+          messagingProtocols: selectedProtocols,
+          sourceContract: '???', // TODO
+          destinationContract: '???', // TODO
+          // imMessage,
+        }
+      }
+      console.log({ swapRequest })
+      swap = await sdk.swap(swapRequest)
+      console.log({ swapResponse: swap })
+
+      if (!!swap.error || swap.resultType !== RoutingResultType.OK) {
+        setError(
+          `Error swapping, routing result: ${swap.resultType}, error: ${swap.error}`
+        )
+        setLoadingSwap(false)
+        return
+      }
+
+      if (fromChain.type === TransactionType.EVM) {
+        const tx = swap.tx as EvmTransaction
+
+        if (tx) {
+          // if approve data is not null, it means approve needed, otherwise it's already approved.
+          if (!!tx?.approveData) {
+            // try to approve
+
+            const result = await signer.signAndSendTx(
+              {
+                type: TransactionType.EVM,
+                blockChain: tx.blockChain?.name,
+                isApprovalTx: !!tx.approveData,
+                from: tx.from,
+                to: tx.txTo,
+                data: tx.txData,
+                value: tx.value,
+                nonce: null,
+                gasLimit: tx.gasLimit,
+                gasPrice: tx.gasPrice,
+                maxPriorityFeePerGas: tx.maxPriorityFeePerGas,
+                maxFeePerGas: tx.maxFeePerGas,
+              },
+              fromAddress,
+              fromChain.chainId
+            )
+            await checkApprovalSync(swap.requestId, result.hash, sdk)
+            console.log('transaction approved successfully')
+          } else {
+            const result = await signer.signAndSendTx(
+              {
+                type: TransactionType.EVM,
+                blockChain: tx.blockChain?.name,
+                isApprovalTx: !!tx.approveData,
+                from: tx.from,
+                to: tx.txTo,
+                data: tx.txData,
+                value: tx.value,
+                nonce: null,
+                gasLimit: tx.gasLimit,
+                gasPrice: tx.gasPrice,
+                maxPriorityFeePerGas: tx.maxPriorityFeePerGas,
+                maxFeePerGas: tx.maxFeePerGas,
+              },
+              fromAddress,
+              fromChain.chainId
+            )
+            const txStatus = await checkTransactionStatusSync(
+              swap.requestId,
+              result.hash,
+              sdk
+            )
+            console.log('transaction finished', { txStatus })
+            console.log('bridged data?', txStatus.bridgeData)
+          }
+        }
+      } else {
+        const tx = swap.tx as CosmosTransaction | Transfer
```

Comment:
Did you mean CosmosTransaction or SolanaTransaction?

Reply:
'rango-types' package does not export any 'SolanaTransaction'.